packages/mds-stream-processor/connectors/kafka-connector.ts

```diff
@@ -58,9 +59,15 @@ export const KafkaSource =
           message: { offset, value }
         } = payload
         if (value) {
-          const message: TMessage = JSON.parse(value.toString())
+          const message: TMessage | ParseError = (() => {
+            try {
+              return JSON.parse(value.toString())
+            } catch (err) {
```

Comment:
Why not catch `SyntaxError` specifically?

Reply:
I'd like to handle _any_ sort of error that could come out, and I'm not sure if SyntaxError is the only kind at the moment. Additionally, tightly constraining this to catching a SyntaxError could present a bug in the future, if JSON.parse() changes what types of errors it can throw, and I don't see why we shouldn't handle any sort of parsing error like this (same net effect).

Reply:
I don't think I've ever seen `.toString()` throw in Javascript; maybe I've just been lucky. Either way, I'm not sure we care _why_ the error is thrown here. We should be able to diagnose based on the message in the dead letter topic.

Reply:
I think that these buffers will away have a `.toString()` representation.